src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py

```diff
@@ -891,7 +893,13 @@ def main():
     if verbose:
         print("creating usgs time_slice data array ...")
 
-    usgs_df = nnu.build_data_assimilation(root, data_assimilation_parameters)
+
```

Comment:
Wrap all of Step 6 into an `if data_assimilation_parameters`, 

then move the rest of the logic into the function, `build_data_assimilation`